### PR TITLE
Add dereferenceable LLVM attribute to some GC functions

### PR DESCRIPTION
**Pull request by gbaraldi**
This may allow for some more optimizations

### CODE REVIEW

File: src/llvm-final-gc-lowering.cpp [hunk 1]
```diff
@@ -208,6 +208,7 @@ Value *FinalLowerGC::lowerGCAllocBytes(CallInst *target, Function &F)
         newI = builder.CreateCall(poolAllocFunc, { ptls, pool_offs, pool_osize });
     }
     newI->setAttributes(newI->getCalledFunction()->getAttributes());
+    newI->addRetAttr(Attribute::getWithDereferenceableBytes(F.getContext(),offset));
```
Comment (Keno) [inline src/llvm-final-gc-lowering.cpp:211]: Does this actually do much? Very few optimizations run after this point (no harm in this of course).

Comment (gbaraldi) [reply src/llvm-final-gc-lowering.cpp:211]: I wonder if that attribute could be added earlier? I will take a look

**Comment by gbaraldi** [comment]
@nanosoldier runbenchmarks(!"scalar", vs=":master")

**Review by vtjnash — APPROVED** [review]

**Comment by gbaraldi** [comment]
I don't think that run ever went through

**Comment by gbaraldi** [comment]
@nanosoldier runbenchmarks(!"scalar", vs=":master")

**Comment by vtjnash** [comment]
@nanosoldier runbenchmarks(!"scalar", vs=":master")

**Comment by vtjnash** [comment]
@nanosoldier `runbenchmarks(!"scalar", vs=":master")`

**Comment by vtjnash** [comment]
@gbaraldi your group membership is private at https://github.com/orgs/JuliaLang/people/gbaraldi, so @nanosoldier will not authenticate you to run jobs (and you need a \``code fence`\`)

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/46256#issuecomment-1238592447) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/463e6a7_vs_31d4c22/report.md).

**Pull request merged**